cmd/trade.go

```diff
@@ -722,6 +726,26 @@ func runTradeCmd(options inputs) {
 	bot.Start()
 }
 
+func getUserID(l logger.Logger, botConfig trader.BotConfig) (string, error) {
+	var userIDPrehash string
+	if botConfig.IsTradingSdex() {
+		userIDPrehash = botConfig.TradingAccount()
+	} else {
+		exchangeAPIKeys := botConfig.ExchangeAPIKeys.ToExchangeAPIKeys()
+		userIDPrehash = exchangeAPIKeys[0].Key
+	}
+
+	// hash avoids exposing the user account or api key
+	userIDHashed, e := utils.HashString(userIDPrehash)
+	if e != nil {
+		logger.Fatal(l, fmt.Errorf("could not create user id: %s", e))
+		return "", e
+	}
+
+	userID := fmt.Sprint(userIDHashed)
+	return userID, e
```

Comment:
error will always be nil here. can `return fmt.Sprint(userIDHashed), nil`
we do this as a convention across the codebase which gives the reader of the code confidence in knowing that all errors have been handled and bubbled up appropriately as soon as they were encountered. It makes the code a little easier to reason about as the codebase gets bigger